### PR TITLE
Support widget-specific preferences

### PR DESCRIPTION
**Pull request by pokey**

### CODE REVIEW

File: widget_preferences.py [hunk 1]
```diff
@@ -87,7 +101,11 @@ def export(self, id:str) -> Dict[str, str]:
         dict[id + '_alignment'] = self.alignment
         dict[id + '_expand_direction'] = self.expand_direction
         dict[id + '_minimized'] = "1" if self.minimized else "0"
-        
+
+        for extra_preference in self.extra_preferences:
```
Comment (chaosparrot) [inline widget_preferences.py:105]: This requires an extra hasattr check - Tested this with my current preferences and this caused a crash.

**Review by chaosparrot — CHANGES_REQUESTED** [review]

File: widget_preferences.py (hunk 1 above)
Comment (pokey) [reply widget_preferences.py:105]: Should we instead add a default value to `ExtraPreference` and set that in the constructor the way we do for the other args?

Comment (chaosparrot) [reply widget_preferences.py:105]: Yeah that could work as well, and seems like a good place to put a default

**Comment by chaosparrot** [comment]
I added a variation of this code in v0.6, so I'm closing this one!

**Comment by pokey** [comment]
Awesome thanks! Sorry to be so incredibly slow on patching up this PR 😅

**Pull request closed without merging**